### ISSUE TITLE
Multistage model as single tflite file

### ISSUE THREAD

**Issue by Deepakpy** [PINTO0309/PINTO_model_zoo#286]
### Issue Type

Support

### OS

Other

### OS architecture

Other

### Programming Language

Python

### Framework

TensorFlowLite

### Model name and Weights/Checkpoints URL

Blazehand model combined with gesture detection model consiting of Fully connected layers

### Description

Microcontroller with 2 to 4x Arm® Cortex®-A53s and 1x Cortex-M4 cores. It brings High-Performance with Low Power, Flexible options of Memory and High-Speed Interfaces as well as advanced Audio and Video capabilities.
OS : Yocto Linux

Issue Description
I am new to mediapipe. I want to use the mediapipe hands (hand detection and hand landmarks models) to be combined along with a simple gesture recognition model. All three to be combined to a single tflite file for inference on a microcontroller.
Steps to Reproduce
Is it possible to take the blaze hand models weights and make those layers non trainable and add few additional FC layers which take two inputs the keypoints from the blaze hand models and additional label input for predefined geustures. The output should be a single prediction of hand gesture.

Any leads will be helpful. To generate the model.
The overall idea is to have this multistage model compressed into a single model which can be trained with just few epochs and with less data for new gestures leaveraging the already trained weights from blaze hand.

**Comment by PINTO0309** [comment]
That's easy.
https://github.com/PINTO0309/simple-onnx-processing-tools
tflite -> onnx -> onnx op generation + onnx merge -> tflite

I can't do anything about it because you have not provided the specific component of `few additional FC layers`. Otherwise, please let me know how you intend to handle multiple bounding boxes of different shapes when multiple hands are detected, resizing them to a fixed size such as 224x224. Padding to a fixed size is easy to come up with, but very redundant. If you can suggest ideas for efficient batching of multiple bounding boxes of different shapes with fixed geometry, I will respond.

In the absence of specific, cool ideas, I don't see much value in this request.

**Comment by PINTO0309** [comment]
- Multi post
https://github.com/vladmandic/human/discussions/287

**Comment by Deepakpy** [comment]
When you said "tflite -> onnx -> onnx op generation + onnx merge -> tflite" does it mean to take both the models (palm detection and hand landmark) convert to onnx and then merge both models into the onnx merge.

I still have not figured out how the palm detect model input can be given to hand landmark detection model.
From the multi post comment you commented inthe last post. The user had adviced me to keep the palm detection model seperately and he suggests to use other techniques to get the hand gestures from the obtained landmarks.
As i found your repository later. I was keen on knowing if it would be possible to merge all three models(palm detection tflite+ keypoints detection model+ custom gesture model which i have posted above) or to merge any of the two models to get less number of tflite files to deal with.
I can't use mediapipe framework on the yocto linux OS and will use the Nnstreamer or gstreamer to utilize the gpu delegate. But as far as i know i can use only one tflite file to run on the gpu or NPU.
Below is the FC layers model which takes the hand keypoints from the hand landmarks model and predicts the guestures.
model = tf.keras.models.Sequential([
    tf.keras.layers.Input((21 * 2, )),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(20, activation='relu'),
    tf.keras.layers.Dropout(0.4),
    tf.keras.layers.Dense(10, activation='relu'),
    tf.keras.layers.Dense(NUM_CLASSES, activation='softmax')
])

**Comment by PINTO0309** [comment]
>I still have not figured out how the palm detect model input can be given to hand landmark detection model.

First, please read all of the logic below and give some thought to whether it is in your best interest to merge the two models as you wish.
https://github.com/metalwhale/hand_tracking

>From the multi post comment you commented inthe last post. The user had adviced me to keep the palm detection model seperately and he suggests to use other techniques to get the hand gestures from the obtained landmarks.
As i found your repository later. I was keen on knowing if it would be possible to merge all three models(palm detection tflite+ keypoints detection model+ custom gesture model which i have posted above) or to merge any of the two models to get less number of tflite files to deal with.

I already understand it all.

**Comment by Deepakpy** [comment]
Thanks for all the reference material. I will get back to you once i properly figure out.

**Comment by PINTO0309** [comment]
Closed due to no activity for over a week.
